crates/sui-adapter-transactional-tests/tests/programmable_transaction_examples/upgrade.move

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// simple test of upgrade
+
+//# init --addresses p=0x0 q=0x0 q_2=0x0 r=0x0 s=0x0 --accounts A
+
+//# publish
+module p::m {
+    public fun foo(x: u64) {
+        p::n::bar(x)
+    }
+}
+module p::n {
+    public fun bar(x: u64) {
+        assert!(x == 1, 0);
+    }
+}
+
+//# publish --upgradeable --sender A
+module q::m {
+    public fun x(): u64 { 0 }
+}
+
+//# publish
+module r::m {
+    public fun y(): u64 { 0 }
+}
+
+//# package
+module q_2::m {
+    public fun x(): u64 { y() + 1 }
+    public fun y(): u64 { r::m::y() }
+}
+
+//# programmable --sender A --inputs 10 @A object(2,1) 0u8
```

Comment:
Would it be too much to create a `digest(_)` input type, like `object(_,_)` that is used to substitute the digest of a staged package? I can imagine that calculating this digest and formatting it this way is otherwise going to be an absolute nightmare.

Reply:
sigh... yeah I should do that